src/main/java/hudson/plugins/mercurial/MercurialSCM.java

```
@@ -927,7 +950,7 @@ private boolean causedByMissingHg(IOException e) {
                 listener.error("Failed to use repository cache for " + getSource(env));
                 return null;
             }
-        } catch (Exception x) {
+        } catch (Throwable x) {
```

should this swallow out of memory errors?